pydra/engine/workers.py

```diff
@@ -246,27 +209,60 @@ def run_el(self, runnable, rerun=False):
         script_dir, batch_script = self._prepare_runscripts(runnable, rerun=rerun)
         if (script_dir / script_dir.parts[1]) == gettempdir():
             logger.warning("Temporary directories may not be shared across computers")
-
         if isinstance(runnable, TaskBase):
-            checksum = runnable.checksum
             cache_dir = runnable.cache_dir
             name = runnable.name
+            uid = runnable.uid
         else:
-            checksum = runnable[-1].checksum_states()[runnable[0]]
             cache_dir = runnable[-1].cache_dir
             name = runnable[-1].name
+            uid = runnable[-1].uid_states()[runnable[0]]
+
+        return self._submit_job(batch_script, name=name, uid=uid, cache_dir=cache_dir)
+
+    def _prepare_runscripts(self, task, interpreter="/bin/sh", rerun=False):
+        if isinstance(task, TaskBase):
+            cache_dir = task.cache_dir
+            ind = None
+            uid = task.uid
+        else:
+            ind = task[0]
+            cache_dir = task[-1].cache_dir
```

Comment:
same comment here with task. what is `task[0]` and `task[-1]` and how long can task be?